PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCacheRenderers/MediaView.swift

```diff
@@ -0,0 +1,288 @@
+//
+//  PBMMediaViewT.swift
+//  PrebidMobileRendering
+//
+//  Copyright © 2021 Prebid. All rights reserved.
+//
+
+import UIKit
+
+ let DEFAULT_VIEWABILITY_POLLING_INTERVAL: TimeInterval = 0.2
+ let IGNORE_CLICKS_IF_UNREGISTERED = true
+
+ enum MediaViewState {
+    case undefined, playbackNotStarted, playing, pausedByUser, pausedAuto, playbackFinished
+}
+
+@objc(PBMMediaView) public class MediaView: UIView, PBMPlayable, PBMAdViewManagerDelegate {
+    
+    @IBInspectable @objc public weak var delegate: PBMMediaViewDelegate?
+    
+    @objc private(set) public var mediaData: MediaData?    // filled on successful load
+    var mediaDataToLoad: MediaData?          // present during the loading
+    
+    var adConfiguration: PBMAdConfiguration!    // created on media loading attempt
+    
+    // TODO: make available only tests {
+    var connection: PBMServerConnectionProtocol?
+    var pollingInterval: NSNumber?  // NSTimeInterval
+    var scheduledTimerFactory: PBMScheduledTimerFactory?
+    // }
+    
+    // Ad loading and management {
+    var vastTransactionFactory: PBMVastTransactionFactory!
+    var adViewManager: PBMAdViewManager?
+    // }
+    
+    // autoPlayOnVisible {
+    @IBInspectable @objc public var autoPlayOnVisible = true {
+        didSet {
+            bindPlaybackToViewability = shouldBindPlaybackToViewability
+        }
+    }
+    
+    var viewabilityPlaybackBinder: PBMViewabilityPlaybackBinder?
+    
+    var bindPlaybackToViewability: Bool {
+        get { self.viewabilityPlaybackBinder != nil }
+        set(bindPlaybackToViewability) {
+            if !bindPlaybackToViewability {
+                // -> turn OFF
+                self.viewabilityPlaybackBinder = nil;
+                return;
+            }
+            if self.viewabilityPlaybackBinder != nil {
+                // already ON
+                return;
+            }
+            // -> turn ON
+            let exposureProvider = PBMViewExposureProviders.visibilityAsExposure(for: self)
+            let pollingInterval = self.pollingInterval?.doubleValue ?? DEFAULT_VIEWABILITY_POLLING_INTERVAL
+            let timerFactory = self.scheduledTimerFactory ?? Timer.pbmScheduledTimerFactory()
+            self.viewabilityPlaybackBinder = PBMViewabilityPlaybackBinder.init(exposureProvider: exposureProvider,
+                                                                               pollingInterval: pollingInterval,
+                                                                               scheduledTimerFactory: timerFactory,
+                                                                               playable: self)
+        }
+    }
+    
+     var shouldBindPlaybackToViewability: Bool {
+        self.autoPlayOnVisible && self.mediaData != nil
+    }
+    // }
+    
+    var state: MediaViewState = .undefined
+    
+    @objc public func loadMedia(_ mediaData: MediaData) {
```

Comment:
please rename to `load( _ mediaData: MediaData)` or `load( with mediaData: MediaData)`